### PR TITLE
ToolFactory resurrection in Planemo

### PR DESCRIPTION
**Pull request by fubar2**
PR* to bring back the ToolFactory (TF) to Planemo so
``planemo tool_factory --galaxy_root ~/galaxy-dev/ --extra_tools ../planemo/testtools/``
brings up a ToolFactory running in Galaxy under planemo at http://localhost:9090

This seems rather wonderful to me. However, I have no idea if planemo needs to be burdened with this extra cruft if the same effect can be had by downloading a local copy and serving it? OTOH it is neat. Please advise. 

The TF has planemo as a requirement and seems to work well inside planemo. Matryoshka dolls come to mind. For devs, the ``--recursive`` flag becomes necessary for a git clone to ensure that the ToolFactory submodule is also installed - not a problem for pip users. Seems better than moving a copy into the Planemo source tree - but irritating because if you forget, nothing works properly.  

Unlike the Docker ToolFactory, there is no history loaded with sample tools and no local toolshed for the round-trip to install new tools back into the Galaxy instance being used. Any suggestions on loading a history and/or a workflow automatically could make it much more useful - being able to rerun the sample tools is very useful for figuring out how it works.

### CODE REVIEW

**Comment by fubar2** [comment]
Ah yes. My old friend flake8 causes tests to fail. 
I'll try to repair all the errors in the ToolFactory repo :)

**Comment by fubar2** [comment]
tl;dr Apropos: #1078 - could there be a version mashing problem using [home]/miniconda for both Galaxy and for Planemo inside a docker container?

This PR runs planemo repeatedly for me when invoked from a shell. I have previously encountered strange problems using TF/Planemo in docker so I  set up a docker container **running this code.** [Dockerfile.planemotf_fails_second_time.txt](https://github.com/galaxyproject/planemo/files/5686475/Dockerfile.planemotf_fails_second_time.txt)
and serving a TF. Planemo runs once then fails repeatedly with a familiar message about gxformat2 problems. Could this gxformat2 version tussle be avoided by having separate minicondas for Galaxy and for Planemo in a docker container?
```
Traceback (most recent call last):
  File "/home/biodocker/miniconda3/envs/mulled-v1-03c4e75023f9966a037f2c006fea55a0cd9e2ab6c0a01e202c3c967d6a0f48a3/bin/planemo", line 6, in <module>
    from planemo.cli import planemo
  File "/home/biodocker/miniconda3/envs/mulled-v1-03c4e75023f9966a037f2c006fea55a0cd9e2ab6c0a01e202c3c967d6a0f48a3/lib/python3.8/site-packages/planemo/cli.py", line 11, in <module>
    from planemo.galaxy import profiles
  File "/home/biodocker/miniconda3/envs/mulled-v1-03c4e75023f9966a037f2c006fea55a0cd9e2ab6c0a01e202c3c967d6a0f48a3/lib/python3.8/site-packages/planemo/galaxy/__init__.py", line 5, in <module>
    from .config import galaxy_config
  File "/home/biodocker/miniconda3/envs/mulled-v1-03c4e75023f9966a037f2c006fea55a0cd9e2ab6c0a01e202c3c967d6a0f48a3/lib/python3.8/site-packages/planemo/galaxy/config.py", line 52, in <module>
    from .workflows import (
  File "/home/biodocker/miniconda3/envs/mulled-v1-03c4e75023f9966a037f2c006fea55a0cd9e2ab6c0a01e202c3c967d6a0f48a3/lib/python3.8/site-packages/planemo/galaxy/workflows.py", line 12, in <module>
    from gxformat2.normalize import inputs_normalized, outputs_normalized
ImportError: cannot import name 'outputs_normalized' from 'gxformat2.normalize' (/home/biodocker/miniconda3/envs/mulled-v1-03c4e75023f9966a037f2c006fea55a0cd9e2ab6c0a01e202c3c967d6a0f48a3/lib/python3.8/site-packages/gxformat2/normalize.py)

```
